source/opt/dominator_analysis_pass.h

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_OPT_DOMINATOR_ANALYSIS_PASS_H_
+#define LIBSPIRV_OPT_DOMINATOR_ANALYSIS_PASS_H_
+
+#include <cstdint>
+#include <map>
+
+#include "dominator_tree.h"
+#include "module.h"
+#include "pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// Interface to perform dominator or postdominator analysis on a given function.
+class DominatorAnalysisBase {
+ public:
+  explicit DominatorAnalysisBase(bool isPostDom) : tree_(isPostDom) {}
+
+  // Calculate the dominator (or postdominator) tree for given function F.
+  inline void InitializeTree(const ir::Function* f) { tree_.InitializeTree(f); }
+
+  // Returns true if BasicBlock A dominates BasicBlock B.
+  inline bool Dominates(const ir::BasicBlock* a,
+                        const ir::BasicBlock* b) const {
+    if (!a || !b) return false;
+    return Dominates(a->id(), b->id());
+  }
+
+  // Returns true if BasicBlock A dominates BasicBlock B. Same as above only
+  // using the BasicBlock IDs.
+  inline bool Dominates(uint32_t a, uint32_t b) const {
+    return tree_.Dominates(a, b);
+  }
+
+  // Returns true if BasicBlock A strictly dominates BasicBlock B.
+  inline bool StrictlyDominates(const ir::BasicBlock* a,
+                                const ir::BasicBlock* b) const {
+    if (!a || !b) return false;
+    return StrictlyDominates(a->id(), b->id());
+  }
+
+  // Returns true if BasicBlock A strictly dominates BasicBlock B. Same as above
+  // only using the BasicBlock IDs.
+  inline bool StrictlyDominates(uint32_t a, uint32_t b) const {
+    return tree_.StrictlyDominates(a, b);
+  }
+
+  // Return the immediate dominator of node A or return NULL if it is an entry
```

Comment:
Minor, but here and elsewhere in this file your comments referring to parameters do not match parameters names. Preferably refer to parameters as |param|. "Returns the immediate dominator of |node| or NULL if it is an entry node."

Also, here uses "Returns".